Sources/AsyncKit/ConnectionPool.swift

```diff
@@ -138,43 +163,56 @@ public final class ConnectionPool<Source> where Source: ConnectionPoolSource  {
     ///
     /// The connection returned by this method should be released when you are finished using it.
     ///
-    ///     let conn = try pool.requestConnection().wait()
+    ///     let conn = try pool.requestConnection(on: ...).wait()
     ///     defer { pool.releaseConnection(conn) }
     ///     // use the connection
     ///
     /// See `withConnection(_:)` for a callback-based method that automatically releases the connection.
     ///
+    /// - parameters:
+    ///     - on: Preferred event loop for the new connection.
     /// - returns: A future containing the requested connection.
-    public func requestConnection() -> EventLoopFuture<Source.Connection> {
-        guard !self.isClosed else {
-            return self.source.eventLoop.makeFailedFuture(ConnectionPoolError.closed)
+    public func requestConnection(
+        on eventLoop: ConnectionPoolEventLoop = .any
+    ) -> EventLoopFuture<Source.Connection> {
+        // synchronize access to available / active connection checks
+        self.lock.lock()
+        defer { self.lock.unlock() }
+
+        let eventLoop = eventLoop.on(self.eventLoopGroup)
+        guard !self.didShutdown else {
+            return eventLoop.makeFailedFuture(ConnectionPoolError.shutdown)
         }
-        
+
         if let conn = self.available.popLast() {
```

Comment:
FYI, using `popLast` here will cause the first element in the event loop to hardly be used when the pool is sufficiently large. That could cause that connection to be closed for inactivity at some point. That might be mitigated by "rotating" through the connections instead, but not sure if there is an O(1) method for that. (See the rest of this comment for a solution that might be better.)

More generally: What if a connection gets closed (e.g. by the server) _while it is not in use_ (i.e. while it is inside `available`)? Wouldn't that cause us to return an already-closed connection in `requestConnection`? That could be avoided by replacing the `popLast` above with a loop that pops a connection, and if it is already, discards it, until it finds a still-open connection? WDYT?